modules/Rock.cmake

```diff
@@ -49,6 +49,30 @@ function(rock_add_compiler_flag_if_it_exists FLAG)
     endif()
 endfunction()
 
+macro(rock_add_target_properties _target _name _properties)
+    set(_new_properties ${_properties} ${ARGN})
+    get_target_property(_old_properties ${_target} ${_name})
+    if(NOT _old_properties)
+        set(_old_properties)
+    else()
+        set(_old_properties "${_old_properties} ")
+    endif()
+    string (REPLACE ";" " " _new_properties "${_old_properties}${_new_properties}")
```

Comment:
It works just the same (except for the APPEND support). If you pass a list of say (-Wone -Wtwo -Wthree) cmake will convert it to "-Wone;-Wtwo;-Wthree" thus breaking compiler's command line.